libvast/src/directory.cpp

```diff
@@ -108,23 +109,18 @@ const path& directory::path() const {
 }
 
 size_t recursive_size(const vast::directory& dir) {
-  size_t size = 0;
-  std::vector<vast::directory> stack{dir};
-  while (!stack.empty()) {
-    auto dir = std::move(stack.back());
-    stack.pop_back();
-    for (auto file : dir) {
-      if (file.is_regular_file()) {
-        if (auto sz = vast::file_size(file)) {
-          VAST_TRACE_SCOPE("{} += {}", file, *sz);
-          size += *sz;
-        }
-      } else if (file.is_directory()) {
-        stack.push_back(vast::directory{file});
-      }
+  size_t total_size = 0;
+
+  const auto p = std::filesystem::path{dir.path().str()};
+  for (const auto& f : std::filesystem::recursive_directory_iterator(p)) {
+    if (f.is_regular_file()) {
+      const auto size = f.file_size();
+      VAST_TRACE_SCOPE("{} += {}", f, size);
```

Comment:
```suggestion
      VAST_TRACE("{} += {}", f, size);
```

While you're at it, please change this as well.

Reply:
Switched to `VAST_TRACE`.